src/main/java/wolox/training/security/CustomAuthenticationService.java

```diff
@@ -0,0 +1,34 @@
+package wolox.training.security;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@Service
+public class CustomAuthenticationService implements UserDetailsService {
+
+	@Autowired
+	private UserRepository userRepository;
+
+	@Override
+	public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+		Optional<User> optional = userRepository.findByUsername(username);
+		if (optional.isEmpty()) {
+			throw new UsernameNotFoundException("Username not found");
+		}
+
+		User u = optional.get();
```

Comment:
You could make this one line by using `User u = userRepository.findByUsername(username).orElseThrow(..)`